Sources/MongoKitten/Cursor.swift

```diff
@@ -34,6 +35,28 @@ internal final class Cursor {
     }
 }
 
+final class CursorDrainer {
```

Comment:
Even though this API is internal, I would prefer this to be `private` in `Cursor`. The API can then be `cursor.drain()` instead of `CursorDrainer(cursor: cursor).collectAll()`, and `CursorDrainer` becomes an implementation detail.